app/src/main/java/com/davidtomas/taskyapp/features/auth/domain/useCase/ValidateInputsUseCase.kt

```diff
@@ -0,0 +1,80 @@
+package com.davidtomas.taskyapp.features.auth.domain.useCase
+
+import com.davidtomas.taskyapp.core.domain.model.Result
+import com.davidtomas.taskyapp.features.auth.domain.model.AllFieldsLoginInputValidationType
+import com.davidtomas.taskyapp.features.auth.domain.model.AllFieldsRegisterValidationType
+import com.davidtomas.taskyapp.features.auth.domain.model.EmailInputValidationType
+import com.davidtomas.taskyapp.features.auth.domain.model.InputValidationError
+import com.davidtomas.taskyapp.features.auth.domain.model.InputValidationErrors
+import com.davidtomas.taskyapp.features.auth.domain.model.InputValidationType
+import com.davidtomas.taskyapp.features.auth.domain.model.PasswordLoginInputValidationType
+import com.davidtomas.taskyapp.features.auth.domain.model.PasswordRegisterInputValidationType
+import com.davidtomas.taskyapp.features.auth.domain.model.UserNameInputValidationType
+import com.davidtomas.taskyapp.features.auth.domain.useCase.ValidatePasswordUseCase.ValidationType.LOGIN
+import com.davidtomas.taskyapp.features.auth.domain.useCase.ValidatePasswordUseCase.ValidationType.REGISTER
+
+class ValidateInputsUseCase(
+    private val validateEmailUseCase: ValidateEmailUseCase,
+    private val validateUserNameUseCase: ValidateUserNameUseCase,
+    private val validatePasswordUseCase: ValidatePasswordUseCase,
+) {
+
+    private val inputValidationError = mutableSetOf<InputValidationError>()
+
+    operator fun invoke(
+        inputValidationType: InputValidationType
+    ): Result<Unit, InputValidationErrors> {
+        inputValidationError.clear()
```

Comment:
What's the point of making this set a member variable of this class and then clearing it? Couldn't you just create the list in the function itself?